### PR TITLE
Treat node:eulerAngle curve as skeletal curve; skeletal blending for common targets

### PR DESCRIPTION
**Pull request by shrinktofit**
Re: cocos-creator/3d-tasks#

Changelog:
 * Curves that would modify `Node.prototype.eulerAngles` are also considered as skeletal curves. So they would be subjected to skeletal animation blending.
 * Common targets that would modify T, R, S, or the `eulerAngles` would be subjected to skeletal animation blending like skeletal curves do.

Request reviews from @YunHsiao  @gameall3d

### CODE REVIEW

File: cocos/core/animation/skeletal-animation-blending.ts [hunk 1]
```diff
@@ -63,11 +63,19 @@ export class BlendStateBuffer {
                 s = scale.value;
                 anyChanged = true;
             }
+
+            // Note: rotation and eulerAngles can not co-exist.
```
Comment (shrinktofit) [inline cocos/core/animation/skeletal-animation-blending.ts:67]: cc @YunHsiao

File: cocos/core/animation/skeletal-animation-blending.ts [hunk 2]
```diff
@@ -165,6 +185,7 @@ interface NodeBlendState {
     properties: {
         position?: PropertyBlendState<Vec3>;
         rotation?: PropertyBlendState<Quat>;
+        eulerAngles?: PropertyBlendState<Vec3>;
         scale?: PropertyBlendState<Vec3>;
```
Comment (shrinktofit) [inline cocos/core/animation/skeletal-animation-blending.ts:189]: @YunHsiao  Please note that only curves that operating  on these properties are subjected to skeletal blending(eg. would call to `setTRS`).

File: cocos/core/animation/skeletal-animation-blending.ts (hunk 1 above)
Comment (YunHsiao) [reply cocos/core/animation/skeletal-animation-blending.ts:67]: Should be more specific on this, it is the two curves that shouldn't coexist, not the properties per se.
And we should add in-editor restrictions on this @gameall3d

File: cocos/core/animation/animation-state.ts [hunk 3]
```diff
@@ -889,22 +915,23 @@ export class AnimationState extends Playable {
     }
 }
 
-function isSkeletonCurve (curve: IRuntimeCurve) {
+function isTargetingTrs (path: TargetPath[]) {
```
Comment (gameall3d) [inline cocos/core/animation/animation-state.ts]: Trs->TRS?

**Review by gameall3d — CHANGES_REQUESTED** [review]

File: cocos/core/animation/skeletal-animation-blending.ts (hunk 2 above)
Comment (YunHsiao) [reply cocos/core/animation/skeletal-animation-blending.ts:189]: That's okay for now

**Review by YunHsiao — APPROVED** [review]

**Review by gameall3d — APPROVED** [review]
LGTM

**Pull request merged**